pypeit/spectrographs/p200_dbsp.py

```diff
@@ -0,0 +1,422 @@
+
+from typing import List
+
+def flip_fits_slice(s: str) -> str:
+    return '[' + ','.join(s.strip('[]').split(',')[::-1]) + ']'
+
+class P200DBSPSpectrograph(spectrograph.Spectrograph):
+    """
+    Child to handle P200/DBSP specific code
+    """
+    ndet = 1
+
+    def __init__(self):
+        # Get it started
+        super(P200DBSPSpectrograph, self).__init__()
+        self.spectrograph = 'p200_dbsp'
+        self.telescope = telescopes.P200TelescopePar()
+
+    def configuration_keys(self):
+        """
+        Return the metadata keys that defines a unique instrument
+        configuration.
+
+        This list is used by :class:`pypeit.metadata.PypeItMetaData` to
+        identify the unique configurations among the list of frames read
+        for a given reduction.
+
+        Returns:
+            list: List of keywords of data pulled from meta
+        """
+        return ['dispname', 'binning', 'dispangle', 'dichroic']
+
+    def init_meta(self):
+        """
+        Generate the meta data dict
+        Note that the children can add to this
+
+        Returns:
+            self.meta: dict (generated in place)
+
+        """
+        meta = {}
+        # Required (core)
+        meta['ra'] = dict(ext=0, card='RA', required_ftypes=['science', 'standard'])
+        meta['dec'] = dict(ext=0, card='DEC', required_ftypes=['science', 'standard'])
+        meta['target'] = dict(ext=0, card='OBJECT')
+
+        meta['dispname'] = dict(ext=0, card='GRATING')
+        meta['decker'] = dict(ext=0, card='APERTURE')
+        meta['binning'] = dict(card=None, compound=True)
+
+        meta['mjd'] = dict(card=None, compound=True)
+        meta['exptime'] = dict(ext=0, card='EXPTIME')
+        meta['airmass'] = dict(ext=0, card='AIRMASS', required_ftypes=['science', 'standard'])
+
+        # Extras for config and frametyping
+        meta['dichroic'] = dict(ext=0, card='DICHROIC')
+        meta['dispangle'] = dict(card=None, rtol=1e-2, compound=True)
+        meta['slitwid'] = dict(ext=0, card='APERTURE')
+        meta['idname'] = dict(ext=0, card='IMGTYPE')
+        # Lamps
+        meta['lampstat01'] = dict(ext=0, card='LAMPS')
+
+        # Ingest
+        self.meta = meta
+
+    def compound_meta(self, headarr: List[fits.Header], meta_key: str):
+        """
+        Methods to generate meta in a more complex manner than simply
+        reading from the header.
+
+        mjd is converted from UTSHUT header
+        dispangle is parsed from ANGLE header
+
+        Args:
+            headarr: List[fits.Header]
+              List of headers
+            meta_key: str
+
+        Returns:
+            value:
+
+        """
+        if meta_key == 'mjd':
+            return Time(headarr[0]['UTSHUT']).mjd
+        elif meta_key == 'dispangle':
+            try:
+                return Angle(headarr[0]['ANGLE']).deg
+            except Exception as e:
+                print(headarr[0]['ANGLE'])
+                raise e
```

Comment:
Should we use a call to `msgs.error()` here before raising the error?

```
msgs.warn("Could not read dispangle from header:" + msgs.newline() + headarr[0]['ANGLE'])
raise e
```

Reply:
nice catch!